New ensure user

omad opened the pull request:
I've rewritten the datacube-ensure-user script into python, and added the ability for it to copy user credentials in `~/.pgpass` for use with the new database server.

I've tested all the functionality myself, but it would be good to get another pair of eyeballs, and test by actual users, before we inflict this on everyone.

omad committed:
Rewrite ensure-user script in python\n\nAdd support for copying a pgpass line to use the same account on a new db host

omad committed:
Fix bugs in datacube-ensure-user.py

omad committed:
Use python datacube-ensure-user in new modules

omad committed:
More logging messages in datacube-ensure-user

omad committed:
Final ensure-user fixes

jeremyh commented on modules/agdc-instances/datacube-ensure-user.py:
```diff
@@ -0,0 +1,185 @@
+        return False
+
+
+def find_credentials(pgpass, host):
```
It's worth adding user as a parameter too (defaulting to current user name?), as it can have multiple entries for different roles on the same host.

(I have separate admin, backup, read-only users configured in my pgpass for the hosts)

jeremyh commented on modules/agdc-instances/datacube-ensure-user.py:
```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
```
Fantastic to see that this is Python now rather than bash 🥇

jeremyh commented on modules/agdc-instances/datacube-ensure-user.py:
```diff
@@ -0,0 +1,185 @@
+    os.umask(0o077)
+    with pgpass.open('w') as dest:
+        for line in filedata:
+            dest.write(line + '\n')
```
We should do this as an atomic file write just in case.

jeremyh reviewed (commented):
Looks good

omad committed:
Improve db-ensure-user

- Support using/connecting as a specific user
- Write .pgpass in single IO action

omad committed:
Atomically write .pgpass in db-ensure-user
